src/pe_reports/data/db_query.py

```diff
@@ -0,0 +1,193 @@
+    try:
+        sql = """SELECT * FROM vw_breach_complete
+        WHERE organizations_uid = %(org_uid)s
+        AND modified_date BETWEEN %(start_date)s AND %(end_date)s"""
+        df = pd.read_sql(
+            sql,
+            conn,
+            params={"org_uid": org_uid, "start_date": start_date, "end_date": end_date},
+        )
+        return df
+    except (Exception, psycopg2.DatabaseError) as error:
+        logging.error(f"There was an problem with your database query {error}")
+    finally:
+        if conn is not None:
+            close(conn)
+
+
+# TODO: update date column in database to be a datetime and add start/end date query
+def query_domMasq(conn, org_uid, start_date, end_date):
+    """Query domain masquerading table."""
+    try:
+        sql = """SELECT * FROM dnstwist_domain_masq
+        WHERE organizations_uid = %(org_uid)s
+        AND date_observed BETWEEN %(start_date)s AND %(end_date)s"""
+        df = pd.read_sql(
+            sql,
+            conn,
+            params={
+                "org_uid": org_uid,
+                "start_date": start_date,
+                "end_date": end_date,
+            },
+        )
+        return df
+    except (Exception, psycopg2.DatabaseError) as error:
+        logging.error(f"There was an problem with your database query {error}")
+    finally:
+        if conn is not None:
+            close(conn)
+
+
+def query_shodan(conn, org_uid, start_date, end_date, table):
+    """Query Shodan Table."""
+    try:
+        sql = """SELECT * FROM %(table)s
+        WHERE organizations_uid = %(org_uid)s
+        AND timestamp BETWEEN %(start_date)s AND %(end_date)s"""
+        df = pd.read_sql(
+            sql,
+            conn,
+            params={
+                "table": AsIs(table),
+                "org_uid": org_uid,
+                "start_date": start_date,
+                "end_date": end_date,
+            },
+        )
+        return df
+    except (Exception, psycopg2.DatabaseError) as error:
+        logging.error(f"There was an problem with your database query {error}")
+    finally:
+        if conn is not None:
+            close(conn)
+
+
+def query_darkweb(conn, org_uid, start_date, end_date, table):
+    """Query Dark Web Table."""
+    try:
+        sql = """SELECT * FROM %(table)s
+        WHERE organizations_uid = %(org_uid)s
+        AND date BETWEEN %(start_date)s AND %(end_date)s"""
+        df = pd.read_sql(
+            sql,
+            conn,
+            params={
+                "table": AsIs(table),
+                "org_uid": org_uid,
+                "start_date": start_date,
+                "end_date": end_date,
+            },
+        )
+        return df
+    except (Exception, psycopg2.DatabaseError) as error:
+        logging.error(f"There was an problem with your database query {error}")
+    finally:
+        if conn is not None:
+            close(conn)
+
+
+def query_darkweb_cves(conn, start_date, end_date, table):
```

Comment:
For `def query_darkweb_cves(conn, start_date, end_date, table):`

`start_date` and `end_date` are not needed for the CSG api call, but could agree they may be needed when pulling data from database for reporting. They are listed as arguments, but not sent in the query.. are they needed?

Reply:
Right, we do not need the start_date/end_data. In report metrics, we take this dataframe and subset it to the 10 most recent cves. I guess we could do that in the query instead if you think that's a better idea.

Reply:
My vote is to remove it for now until is is a requirement to produce the cve metric then.

Reply:
@schmelz21 
I have removed these two parameters per commit [9fc5ae1](https://github.com/cisagov/pe-reports/pull/95/commits/9fc5ae1ff77aed606c7ce5a4a16e91ce3a0f6a96)